AutoRouteManager.php

```diff
@@ -64,20 +71,43 @@ public function __construct(
      */
     public function buildUriContextCollection(UriContextCollection $uriContextCollection)
     {
-        $this->getUriContextsForDocument($uriContextCollection);
+        $this->collectionBuilder->build($uriContextCollection);
 
         foreach ($uriContextCollection->getUriContexts() as $uriContext) {
-            $existingRoute = $this->adapter->findRouteForUri($uriContext->getUri(), $uriContext);
+            $translatedSubject = null;
+            $subject = $uriContextCollection->getSubjectObject();
 
-            $autoRoute = null;
+            if (null !== $uriContext->getLocale()) {
+                $translatedSubject = $this->adapter->translateObject($subject, $uriContext->getLocale());
+
+                if (null === $translatedSubject) {
+                    @trigger_error('AdapterInterface::translateObject() has to return the subject as of version 1.1, support for by reference will be removed in 2.0.', E_USER_DEPRECATED);
```

Comment:
Shall we keep it for now? It leaves us then with the option of releasing it in 1.x - I would also like to introduce Doctrine ORM support in 2.0 and I don't know if that will require BC breaking changes.